cpp/include/cudf/jit/span.cuh

```diff
@@ -97,5 +98,52 @@ struct device_span {
   CUDF_HOST_DEVICE [[nodiscard]] constexpr element_type* end() const { return _data + _size; }
 };
 
+/**
+ * @brief A span type with optional elements.
+ *
+ */
```

Comment:
```suggestion
/**
 * @brief A span type with optional elements.
 */
```

Reply:
What does optional mean here? I don't see any optional.

Reply:
That is not similar to an optional. For null, the array element still exists, and it is a null instead of not available as an optional. I think the better name should be something like `nullable_span` or `span_with_null` or something else.

Reply:
The `optional` is the correct term for libcudf. It is inline with optional-iterators we use for identifying null rows in a column. https://github.com/rapidsai/cudf/blob/c9231de061967575b293de03863c4d55e1e885eb/cpp/include/cudf/column/column_device_view.cuh#L362

Reply:
If want to keep the current name, please add more comment to clarify what it is. Just "optional elements" seems very confusing.

Reply:
done. 


Yeah, I understand. I suppose that's because of its internal representation. Most would assume it is a `span<std::optional>`, it's just like `std::vector<bool>`